### PR TITLE
Round parsing and first attempt at creating highlight time ranges

### PR DESCRIPTION
**Pull request by tjlam**
- Created Round class
- Parsed demo file into rounds
- Parsed key events in each round using the high-rate-of-events model 
- outputs time ranges for each round that involve a high rate of events 

Sorry for the long PR.

### CODE REVIEW

File: demofile-parsing/parse_demo.js [hunk 1]
```diff
@@ -1,24 +1,119 @@
...
+  var mins = Math.floor(seconds/60);
+  seconds = Math.floor(seconds%60);
+  hours = hours < 10 ? `0${hours}` : hours;
+  mins = mins < 10 ? `0${mins}` : mins;
+  seconds = seconds < 10 ? `0${seconds}` : seconds;
+  return `${hours}h${mins}m${seconds}s`;
+}
+
+
+fs.readFile(path.resolve(b), (err, buffer) => {
```
Comment (mattlianje) [inline demofile-parsing/parse_demo.js]: This line is blowing up for me....I checked with filepaths that are working fine with me on the previous version. A bit of a mystery. Not sure if it is a windows thing? @Eldrick19 works for you?

File: demofile-parsing/parse_demo.js [hunk 2]
```diff
@@ -1,24 +1,119 @@
 const fs = require("fs");
 const demofile = require("demofile");
 const path = require("path");
+const Round = require('./Round.js');
 
-fs.readFile(path.resolve("/Users/tylerlam/Downloads/test.dem"), (err, buffer) => {
+const DEBUG = false;
```
Comment (mattlianje) [inline demofile-parsing/parse_demo.js]: @tjlam Is this a JS y thing that will flip the switch and become true if something hits the fan?

File: demofile-parsing/Round.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      return -1;
+    }
+    if ( a.time > b.time ){
+      return 1;
+    }
+    return 0;
+  }
+
+  calculateEventRates() {
+    var i;
```
Comment (mattlianje) [inline demofile-parsing/Round.js]: Pedantic - but this lonesome i outside the for loop is stressing me.

File: demofile-parsing/parse_demo.js (hunk 1 above)
Comment (ewega) [reply demofile-parsing/parse_demo.js]: Fixed with Lam's "change b to demPath". Should that not be a permanent change though?

File: demofile-parsing/parse_demo.js [hunk 4]
```diff
@@ -1,24 +1,119 @@
...
+      round.mergeTimeRanges();
+      round.mapToStreamTimestamps(demoGameStart, streamGameStart);
+      // reset roundEvents 
+      roundEvents = [];
+      // log round end
+      console.log(`round ${roundIndex} ended ${secondsToTimestamp(t, demoGameStart, streamGameStart)}`);
+      if (DEBUG) {
+        round.plotAllEvents();
+        round.plotHighRates();
+        round.plotHighlightTimes();
```
Comment (ewega) [inline demofile-parsing/parse_demo.js:70]: Getting timestamps but not any ASCII Graph plotted. @mattlianje You get anything?

Comment (ewega) [reply demofile-parsing/parse_demo.js:70]: Nevermind I'm a dumbo, it seems DEBUG isn't ever returned as true. Manual testing I'm guessing

File: demofile-parsing/parse_demo.js (hunk 2 above)
Comment (ewega) [reply demofile-parsing/parse_demo.js]: I think you have to manually flip it to true for testing purposes. No where in the code it returns to true.

File: demofile-parsing/Round.js (hunk 3 above)
Comment (ewega) [reply demofile-parsing/Round.js]: the boyboy var i

File: demofile-parsing/parse_demo.js (hunk 1 above)
Comment (ewega) [inline demofile-parsing/parse_demo.js]: Everything's good if we could just change this "b" variable to "demPath" for consistency.

**Review by ewega — DISMISSED** [review]
Everything's good, if we could just change the "b" variable to be "demPath I think should be good

File: demofile-parsing/parse_demo.js (hunk 1 above)
Comment (tjlam) [reply demofile-parsing/parse_demo.js]: Yee thats my bad 😬

File: demofile-parsing/parse_demo.js (hunk 4 above)
Comment (tjlam) [reply demofile-parsing/parse_demo.js:70]: Set `DEBUG = true` on line 6 and you should see some graphy bois pop up

**Review by mattlianje — APPROVED** [review]
Took this bad boy for a spin and works like a charm.

**Review by ewega — APPROVED** [review]
Im good

**Pull request merged**